craft_providers/lxd/launcher.py

```diff
@@ -72,48 +105,96 @@ def _create_instance(
+            _set_timezone(
+                base_instance,
+                base_instance.project,
+                base_instance.remote,
+                base_instance.lxc,
+            )
+            base_instance.config_set("user.craft_providers.status", "FINISHED")
+            # set the full instance name as image description
+            lxc.config_set(
+                instance_name=base_instance.instance_name,
+                key="image.description",
+                value=base_instance.name,
+                project=project,
+                remote=remote,
+            )
+            config_timer.stop()
+            base_instance.stop()
+
+        # Copy the base instance to the instance.
+        logger.info("Creating new instance from base instance")
+        logger.debug(
+            "Creating new instance %r from base instance %r",
+            instance.instance_name,
+            base_instance.instance_name,
         )
         lxc.copy(
-            source_remote=remote,
-            source_instance_name=instance.instance_name,
+            source_remote=base_instance.remote,
+            source_instance_name=base_instance.instance_name,
             destination_remote=remote,
-            destination_instance_name=base_instance.instance_name,
+            destination_instance_name=instance.instance_name,
             project=project,
         )
-
-        # set the full instance name as image description
-        lxc.config_set(
-            instance_name=base_instance.instance_name,
-            key="image.description",
-            value=base_instance.name,
-            project=project,
-            remote=remote,
+        _set_timezone(instance, project, remote, lxc)
+    else:
+        instance.launch(
+            image=image_name, image_remote=image_remote, ephemeral=ephemeral
         )
+        instance_status = instance.config_get("user.craft_providers.status")
+
+        # Skip the base configuration if the instance is already configured.
+        if instance_status != "FINISHED":
+            logger.info("Setting up instance")
+            logger.debug("Setting up instance %r", instance.instance_name)
+            instance.config_set("user.craft_providers.status", "PREPARING")
+            config_timer = InstanceTimer(instance)
+            config_timer.start()
+            base_configuration.setup(executor=instance)
+            _set_timezone(instance, project, remote, lxc)
+            instance.config_set("user.craft_providers.status", "FINISHED")
+            config_timer.stop()
+            if not ephemeral:
+                # stop ephemeral instances will delete them immediately
+                instance.stop()
 
     # after creating the base instance, the id map can be set
     if map_user_uid:
         _set_id_map(instance=instance, lxc=lxc, project=project, remote=remote, uid=uid)
 
     # now restart and wait for the instance to be ready
-    instance.start()
+    if ephemeral:
```

Comment:
Nice, I think you may have fixed a bug with stopping and starting ephemeral instances.